pkg/util/deployment/deployment.go

```diff
@@ -288,22 +288,30 @@ func GetActualReplicaCountForReplicaSets(replicaSets []*extensions.ReplicaSet) i
 }
 
 // Returns the number of available pods corresponding to the given replica sets.
-func GetAvailablePodsForReplicaSets(c clientset.Interface, rss []*extensions.ReplicaSet, minReadySeconds int32) (int32, error) {
-	allPods, err := GetPodsForReplicaSets(c, rss)
+func GetAvailablePodsForReplicaSets(c clientset.Interface, deployment *extensions.Deployment, rss []*extensions.ReplicaSet, minReadySeconds int32) (int32, error) {
+	podList, err := listPods(deployment, c)
+	if err != nil {
+		return 0, err
+	}
+	return FindAvailablePodsForReplicaSets(rss, minReadySeconds, podList)
+}
+
+func FindAvailablePodsForReplicaSets(rss []*extensions.ReplicaSet, minReadySeconds int32, podList *api.PodList) (int32, error) {
+	allPods, err := GetPodsForReplicaSets(rss, podList)
```

Comment:
Under what conditions will allPods != podList?  This would mean there is a Pod that matches the deployment that does not belong in a RS?